### ISSUE TITLE
Pad operation modifies labels

### ISSUE THREAD

**Issue by glupol** [fepegar/torchio#626]
**🐛Bug**
The Affine transformation modifies labels at the edges.

**To reproduce**

```python
import torchio as tio
import torch
a = torch.tensor([[1, 0, 0],[0, 1, 0],[0, 0, 1]]).reshape(1, 3, 3, 1)
s = tio.Subject(lab=tio.LabelMap(tensor=a))
cp = tio.RandomAffine(translation = (0, 1, 1), default_pad_value='otsu')
sb = cp(s)
print(sb.lab.data)

```

**Expected behavior**

The augmented map is now corrupted at the edges after the translation.
The default padding value should always be 0 for labels.

**Actual behavior**

At the moment the values from outside of the original data range are set to some non-zero value.
This is expected for the non-label maps, but is a corruption of data for the labels.

**Comment by fepegar** [comment]
Good catch! It should be fixed in #627.
